### DISCUSSION TITLE
Nested selects in livewire 3

### DISCUSSION THREAD

**Discussion by alaraco1** [livewire/livewire#8044] [1. Help]
Hello, I am working on a sales system, and I am doing a nested select between -> families->categories->subcategories. but when doing the edid function in my controller it is not possible to select correctly in category.

this is my subcategory controller: 
 `
class SubcategoryController extends Component
    {use WithPagination;
    public $search;
    public $cant=10;
    public $selected_id;
    public $modal=false;

        public $families;
    public $categories = [];
    public $subcategories = [];
    public $selectedFamily = null;
    public $selectedCategory = null;
    public $subcategoryName = '';

    

    protected $listeners=['delete'];

    public function updatingSearch()
    {
        $this->resetPage();
    }
    public function updatingCant()
    {
        $this->resetPage();
    }
    public function mount(){

            $this->families=Family::all();
    }

    public function render()
    {

        $subcategoriesAll=Subcategory::where('id', 'like', '%'  . $this->search .'%')
        ->orWhere('name', 'like', '%'  . $this->search .'%')
        ->orWhereHas('category', function (Builder $query) {
            $query->where('name', 'like', '%'  . $this->search .'%' );
           })
        ->latest('id')
        ->paginate($this->cant);

        

        return view('livewire.admin.manage.subcategory-controller',compact('subcategoriesAll'/* ,'categories','families' */) )->layout('layouts.admin') ;
    }

    public function updatedSelectedFamily($family_id)
    {
        $this->categories = Category::where('family_id', $family_id)->get();
        $this->selectedCategory = null;
        $this->subcategories = [];
    }

    public function updatedSelectedCategory($category_id)
    {
        $this->subcategories = Subcategory::where('category_id', $category_id)->get();
    }

    public function create(){
        $this->validate([
            'selectedFamily'=>'required|exists:families,id',
            'selectedCategory'=>'required|exists:categories,id',
            'subcategoryName'=>'required|unique:subcategories,name',
        ]);

        Subcategory::create([
            'category_id'=>$this->selectedCategory,
            'name'=>$this->subcategoryName,
        ]);

      
        $this->closemodal();
        $this->dispatch('success','Subcategoria creada con exito');
    }

 

    public function closemodal()
    {   
        $this->reset([
            'modal',
            'selectedFamily',
                'selectedCategory',
                'subcategoryName',
                            'name',
                            'selected_id'
        ]);
        $this->resetValidation();
    }

    public function edit(Subcategory $subcategory){

        $record=$subcategory;
        $this->selected_id=$record->id;
        $this->selectedFamily=$record->category->family_id;
        $this->selectedCategory=$record->category_id;
        $this->subcategoryName=$record->name;
        $this->modal=true;
        $this->updatedSelectedFamily($this->selectedFamily);
        $this->updatedSelectedCategory($this->selectedCategory);

    }   
   
}
`

this is my modal view

`
      
<div >

          <div class="mb-4" >
       
            <x-label class="block text-sm font-medium text-gray-700 mb-2">Familias: </x-label>

                      <select  wire:model.live="selectedFamily"  class="w-full border-gray-300 mr-2 dark:border-gray-700 dark:bg-gray-900 dark:text-gray-300 focus:border-indigo-500 dark:focus:border-indigo-600 focus:ring-indigo-500 dark:focus:ring-indigo-600 rounded-md shadow-sm">
                          <option value="" disabled>Seleccione...</option>
                          @foreach ($families as $family)
                                <option value="{{$family->id}}">{{ $family->name }}</option>
                            @endforeach
              
                                                      
                      </select> 
                      
            
            </div>
        
            <div class="mb-4" >

              <x-label class="block text-sm font-medium text-gray-700 mb-2">Categoria  </x-label>
  
                        <select  wire:model.live="selectedCategory"   class="w-full border-gray-300 mr-2 dark:border-gray-700 dark:bg-gray-900 dark:text-gray-300 focus:border-indigo-500 dark:focus:border-indigo-600 focus:ring-indigo-500 dark:focus:ring-indigo-600 rounded-md shadow-sm">
                            <option value="" disabled>Seleccione...</option>
                                @if (!is_null($selectedFamily))
                                @foreach ($categories as $category)
                                    <option value="{{ $category->id }}">{{ $category->name }}</option>
                                @endforeach
                            @endif
                   
                                                        
                        </select> 

          
              </div>

               <div >
                <x-label for="name" class="mb-2" >
                  Nombre
                </x-label>
                <x-input wire:model="subcategoryName"  type="text" id="name"  class="w-full"/>
                 
                </div>
              
</div>
     
`
Only the first one is always selected and not the one that should be, already check and the ids if they are loaded normally only that they are not selected in the view

![image](https://github.com/livewire/livewire/assets/161614971/dd889c57-7e08-4553-b8fd-2b0af6601cc1)

**Comment by lrljoe** [comment]
I suggest that you share a video example of what is happening, as your explanation isn't wholly clear

Please always format your code when sharing as it makes it easier to help.

A disabled option to use as "empty" won't be selected by the way, so if it's the problem that the first item is always checked, then that may be the cause , would need you to share the actual issue to know more
